common/men_at_arms_types/00_maa_types.txt

```diff
@@ -126,6 +127,35 @@ pikemen_unit = {
 	icon = pikemen
 }
 
+caster_unit = {
+	type = caster
+	
+	damage = 30
+	toughness = 9
+	pursuit = 5
+	screen = 0
+	
+	terrain_bonus = {
+		plains = { damage = 10 pursuit =10 }
+		drylands =  { damage = 10 pursuit =15 }
+		mountains = { damage = 14 pursuit = 14 }
+		hills = { damage = 8 toughness = 4 pursuit = 12}
+	}
+
+	counters = {
+		massive_aerial = 1
+		light_aerial = 1
+		massive_infantry =1
+	}
+
+	buy_cost = { gold = caster_recruitment_cost }
+	low_maintenance_cost = { gold = caster_low_maint_cost }
+	high_maintenance_cost = { gold = caster_high_maint_cost }
+	
+	stack = 100
+	ai_quality = { value = culture_ai_weight_pikemen }
```

Comment:
But... they're not pikemen, they're casters.

Reply:
Did not do this yet, it will take a lot more time, please check the rest.